relayer/app/cursors/cursors.proto

```diff
@@ -0,0 +1,24 @@
+syntax = "proto3";
+
+package relayer.app.cursors;
+
+import "cosmos/orm/v1/orm.proto";
+
+option go_package = "relayer/app";
+
+message Cursor {
+  option (cosmos.orm.v1.table) = {
+    id: 1;
+    primary_key: { fields: "src_chain_id,conf_level,dst_chain_id,attest_offset" }
+    index: { id: 1, fields: "confirmed" }
+    index: { id: 2, fields: "src_chain_id,conf_level,dst_chain_id" }
+  };
+
+  uint64 src_chain_id       = 1; // Chain ID as per https://chainlist.org
+  uint64 dst_chain_id       = 2; // Chain ID as per https://chainlist.org
+  uint32 conf_level         = 3; // Confirmation level of the cross-chain block
+  uint64 attest_offset      = 4; // Offset of the cross-chain block
+  uint64 first_xmsg_offset  = 5; // Offset of the first cross-chain message
+  uint64 last_xmsg_offset   = 6; // Offset of the last cross-chain message
```

Comment:
ah right, so this is not going to work. Multiple shards map to the same conf level. So we must store something like: `map[ShardID]map[XMsgRange]`. 

My suggestion, lets skip this xmsgOffset stuff, lets stick to original `bool empty`.
```
empty := Attestation > XBlock > Xmsgs > filter by shard ConfLevel > empty or not
confirmed := SubmitCursor(FinalizedRef).AtterstOffset >= AttestOffset
```

Reply:
Changed that now back, the issue I have is that cursors now don't persist message offset which is needed by the message filter https://github.com/omni-network/omni/blob/7fb91f4cf778e0b586aa85f0aa8098ab199c41f3/relayer/app/worker.go#L112

Reply:
I believe we still need to persist last msg offset